src/core/utils/retry.ts

```diff
@@ -0,0 +1,49 @@
+import { randomBytes } from 'crypto'
+import debug from 'debug'
+import { promisify } from 'util'
+
+const log = debug('settlement-core')
+
+/** Wait and resolve after the given number of milliseconds */
+export const sleep = (ms: number) => new Promise(r => setTimeout(r, ms))
+
+/** Generate a floating-point, pseudo-random number in the range [0, 1) */
+export const generateRandom = async () =>
+  (await promisify(randomBytes)(4)).readUInt32BE(0) / 4294967296
+
+const RETRY_MAX_ATTEMPTS = 16
```

Comment:
Should this be configurable?

Reply:
I updated the spec the other day to mandate clients to retry indefinitely (applies to both sending settlements and crediting incoming settlements). I think this is necessary for it to be safe, in case there was a network error on the last retry, which might result in crediting double amounts. Thoughts?